src/Microsoft.DotNet.Build.Tasks/PrereleaseResolveNuGetPackageAssets.cs

```diff
@@ -664,27 +667,228 @@ private void TryParseRuntimeVersion(string imageRuntimeVersion, out bool isWinMD
         /// <returns>The CLR runtime version or empty if the path does not exist.</returns>
         private static string TryGetRuntimeVersion(string path)
         {
-            StringBuilder runtimeVersion = null;
-            uint hresult = 0;
-            uint actualBufferSize = 0;
-            int bufferLength = 11; // 11 is the length of a runtime version and null terminator v2.0.50727/0
+            try
+            {
+                using (FileStream stream = File.OpenRead(path))
+                using (PEReader peReader = new PEReader(stream))
+                {
+                    return peReader.GetMetadataReader().MetadataVersion;
+                }
+            }
+            catch (Exception)
+            {
+                return string.Empty;
+            }
+        }
+    }
 
-            do
+    /// <summary>
+    /// Managed implementation of a reader for getting the runtime version of an assembly
+    /// </summary>
+    static class ManagedRuntimeVersionReader
```

Comment:
I assume this is no longer needed after you switched to using the metadata reader.